aquarius/events/veAllocate.py

```diff
@@ -0,0 +1,82 @@
+#
+# Copyright 2021 Ocean Protocol Foundation
+# SPDX-License-Identifier: Apache-2.0
+#
+import json
+import logging
+import os
+from datetime import datetime
+
+import elasticsearch
+import requests
+
+from aquarius.events.util import make_did
+
+logger = logging.getLogger(__name__)
+
+
+class VeAllocate:
+    def __init__(self, es_instance):
+        self.update_time = None
+        self._es_instance = es_instance
+
+    def retrieve_new_list(self, env_var):
+        """
+        :param env_var: Url of the file containing purgatory list.
+        :return: Object as follows: {...('<did>', '<reason>'),...}
+        """
+        response = requests.post(os.getenv(env_var))
+
+        if response.status_code == requests.codes.ok:
+            logger.info(
+                f"veAllocate: Successfully retrieved list from {env_var} env var."
+            )
+            return {
+                (a["nft_addr"], a["ve_allocated"], a["chainID"])
+                for a in response.json()
+                if a and "nft_addr" in a and "ve_allocated" in a and "chainID" in a
```

Comment:
I suggest `if all(k in a for k in ("nft_addr", "ve_allocated", "chainId"))` for concision, but both work :)